packages/backend-for-frontend/test/inAppNotificationApiConverter.test.ts

```diff
@@ -0,0 +1,650 @@
+        portachiavi: 0,
+        totalCount: 10,
+      },
+      fruizione: {
+        richieste: 0,
+        finalita: 5, // only purposeActivatedRejectedToConsumer
+        totalCount: 5,
+      },
+      "catalogo-e-service": {
+        totalCount: 3,
+      },
+      aderente: {
+        deleghe: 7, // only delegationApprovedRejectedToDelegator
+        anagrafica: 0,
+        totalCount: 7,
+      },
+      "gestione-client": {
+        "api-e-service": 0,
+        totalCount: 0,
+      },
+      totalCount: 25,
+    };
+
+    expect(result).toEqual(expected);
+  });
+
+  it("should handle notification types with zero values", () => {
+    const input: inAppNotificationApi.NotificationsByType = {
+      totalCount: 15,
+      results: {
+        agreementManagementToProducer: 0,
+        agreementSuspendedUnsuspendedToProducer: 5,
+        clientAddedRemovedToProducer: 0,
+        purposeStatusChangedToProducer: 10,
+        templateStatusChangedToProducer: 0,
+      },
+    };
+
+    const result = toBffApiNotificationsCountBySection(input);
+
+    const expected: bffApi.NotificationsCountBySection = {
+      erogazione: {
+        richieste: 5, // 0 + 5
+        finalita: 10, // 0 + 10
+        "template-eservice": 0,
+        "e-service": 0,
+        portachiavi: 0,
+        totalCount: 15,
+      },
+      fruizione: {
+        richieste: 0,
+        finalita: 0,
+        totalCount: 0,
+      },
+      "catalogo-e-service": {
+        totalCount: 0,
+      },
+      aderente: {
+        deleghe: 0,
+        anagrafica: 0,
+        totalCount: 0,
+      },
+      "gestione-client": {
+        "api-e-service": 0,
+        totalCount: 0,
+      },
+      totalCount: 15,
+    };
+
+    expect(result).toEqual(expected);
+  });
+
+  it("should handle shared notification types correctly", () => {
+    // clientKeyAddedDeletedToClientUsers appears in both erogazione.portachiavi and gestione-client.api-e-service
+    const input: inAppNotificationApi.NotificationsByType = {
+      totalCount: 20,
+      results: {
+        clientKeyAddedDeletedToClientUsers: 20,
+      },
+    };
+
+    const result = toBffApiNotificationsCountBySection(input);
+
+    const expected: bffApi.NotificationsCountBySection = {
+      erogazione: {
+        richieste: 0,
+        finalita: 0,
+        "template-eservice": 0,
+        "e-service": 0,
+        portachiavi: 20,
+        totalCount: 20,
+      },
+      fruizione: {
+        richieste: 0,
+        finalita: 0,
+        totalCount: 0,
+      },
+      "catalogo-e-service": {
+        totalCount: 0,
+      },
+      aderente: {
+        deleghe: 0,
+        anagrafica: 0,
+        totalCount: 0,
+      },
+      "gestione-client": {
+        "api-e-service": 20, // same notification type used here too
+        totalCount: 20,
+      },
+      totalCount: 20,
+    };
+
+    expect(result).toEqual(expected);
+  });
+});
+
+describe("calculateDeepLink (via toBffApiNotifications)", () => {
```

Comment:
maybe we could split the converter functions in two files, and split the test here, for readability